### PR TITLE
Update ginkgo to v2

### PR DESCRIPTION
**Pull request by arielireni**
Updated to v2 in order to use JustAfterEach and OrderedContext.
The PR includes also changes made after `go mod tidy` & `go mod vendor`.

Signed-off-by: arielireni <aireni@redhat.com>

### CODE REVIEW

**Review by gciavarrini — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -23,7 +23,8 @@ require (
 	github.com/kube-object-storage/lib-bucket-provisioner v0.0.0-20210818162813-3eee31c01875
 	github.com/morikuni/aec v1.0.0 // indirect
 	github.com/onsi/ginkgo v1.16.5
```
Comment (jakub-dzon) [inline go.mod:25]: Why do we keep v1 dependency?

File: controllers/suite_test.go [hunk 2]
```diff
@@ -47,37 +45,42 @@ var testEnv *envtest.Environment
...
-	Expect(err).NotTo(HaveOccurred())
 
-	err = obv1.AddToScheme(scheme.Scheme)
-	Expect(err).NotTo(HaveOccurred())
-
-	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
-	Expect(err).NotTo(HaveOccurred())
-}, 60)
+	done := make(chan interface{})
+	go func() {
```
Comment (masayag) [inline controllers/suite_test.go:55]: Isn't there a risk that this function will be executed only after the test already started?

Comment (masayag) [reply controllers/suite_test.go:55]: Unless `Eventually` enforces either its completion or its timeout :-)

**Comment by eloycoto** [comment]
For the Kind job, looks like that the issue is that Go is not installed with the correct version, so I'll add the following to the action:

 
```
diff --git a/.github/workflows/kind.yaml b/.github/workflows/kind.yaml
index 38c82f4..b5e35df 100644
--- a/.github/workflows/kind.yaml
+++ b/.github/workflows/kind.yaml
@@ -10,6 +10,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@master
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+
       - uses: engineerd/setup-kind@v0.5.0
         with:
           version: "v0.11.1"
```

So you're making sure that it's 1.16 as code.

**Review by masayag — APPROVED** [review]

**Comment by gciavarrini** [comment]
Good job @arielireni! :)

File: controllers/suite_test.go (hunk 2 above)
Comment (arielireni) [reply controllers/suite_test.go:55]: yes it is, as described [here](https://onsi.github.io/gomega/#eventually)

**Pull request merged**